virttest/utils_libvirt/libvirt_ceph_utils.py

```diff
@@ -0,0 +1,173 @@
+    key_opt = ""
+    is_local_img_file = True if img_file is None else False
+    rbd_key_file = None
+
+    # Prepare a blank params to confirm if delete the configure at the end of the test
+    ceph_cfg = ""
+    disk_auth_dict = None
+    auth_sec_uuid = None
+    if not utils_package.package_install(["ceph-common"]):
+        raise exceptions.TestError("Failed to install ceph-common")
+
+    # Create config file if it doesn't exist
+    ceph_cfg = ceph.create_config_file(ceph_mon_ip)
+    # If enable auth, prepare a local file to save key
+    if ceph_client_name and ceph_client_key:
+        with open(key_file, 'w') as f:
+            f.write("[%s]\n\tkey = %s\n" %
+                    (ceph_client_name, ceph_client_key))
+        key_opt = "--keyring %s" % key_file
+        rbd_key_file = key_file
+    if is_setup:
+        # If enable auth, prepare disk auth
+        if ceph_client_name and ceph_client_key:
+            auth_sec_uuid = _create_secret(auth_sec_usage_type, ceph_auth_key)
+            disk_auth_dict = {"auth_user": ceph_auth_user,
+                              "secret_type": auth_sec_usage_type,
+                              "secret_uuid": auth_sec_uuid}
+        # clean up image file if exists
+        ceph.rbd_image_rm(ceph_mon_ip, ceph_disk_name.split('/')[0],
+                          ceph_disk_name.split('/')[1], keyfile=rbd_key_file)
+
+        #Create necessary image file if not exists
+        _create_image(device_format, img_file, vm.name, storage_size,
+                      ceph_disk_name, ceph_mon_ip, key_opt,
+                      ceph_auth_user, ceph_auth_key)
+
+        # Disk related config
+        disk_src_dict = {"attrs": {"protocol": "rbd",
+                                   "name": ceph_disk_name},
+                         "hosts":  [{"name": ceph_mon_ip,
+                                     "port": ceph_host_port}]}
+        # Create network disk
+        disk_xml = libvirt_disk.create_primitive_disk("network", device, device_target, device_bus,
+                                                      device_format, disk_src_dict, disk_auth_dict)
+        if not keep_raw_image_as:
+            if hotplug:
+                attach_option = params.get("virt_device_attach_option", "--live")
+                virsh.attach_device(vm.name, disk_xml.xml,
+                                    flagstr=attach_option, ignore_status=False, debug=True)
+            else:
+                vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
+                vmxml.add_device(disk_xml)
+                vmxml.sync()
+    else:
+        ceph.rbd_image_rm(ceph_mon_ip, ceph_disk_name.split('/')[0],
+                          ceph_disk_name.split('/')[1], keyfile=rbd_key_file)
+        # Remove ceph config file if created.
+        if os.path.exists(ceph_cfg):
+            os.remove(ceph_cfg)
+        # Remove ceph key file if has.
+        if os.path.exists(key_file):
+            os.remove(key_file)
```

Comment:
I'd like to suggest optionally you avoid code duplication, you're running `if ..exists: ..remove` twice here.

Reply:
Do you means combine two files deletion together? I have updated code as below
```
        # Remove ceph config and key file if created.
        for file_path in [ceph_cfg, key_file]:
            if os.path.exists(file_path):
                os.remove(file_path)
```